File: Week Assignment 6 using C/7_interleave_queue.c
#include <stdio.h>
#define MAX 1000

int queue[MAX];
int front = 0, rear = -1;

void enqueue(int x) { queue[++rear] = x; }
int dequeue() { return queue[front++]; }

void interleaveQueue(int n) {
    int half = n / 2;
    int first[MAX], top1 = -1;
    for (int i = 0; i < half; i++) first[++top1] = dequeue();
    for (int i = 0; i < half; i++) {
        enqueue(first[i]);
        enqueue(queue[front++]);
    }
}

File: Week Assignment 6 using Python/7_interleave_queue.py
from queue import Queue

def interLeaveQueue(q):
    n = q.qsize()
    half = n // 2
    stack = []
    for _ in range(half):
        stack.append(q.get())
    while stack:
        q.put(stack.pop())
    for _ in range(half):
        q.put(q.get())
    for _ in range(half):
        stack.append(q.get())
    while stack:
        q.put(stack.pop())
        q.put(q.get())
    return q
